src/provider/provider_os_memory_linux.c

```diff
@@ -125,6 +131,7 @@ int os_mmap_aligned(void *hint_addr, size_t length, size_t alignment, int prot,
     size_t page_size = os_get_page_size();
 
     if (alignment > page_size) {
+        assert_is_page_aligned(alignment, page_size);
```

Comment:
I think we should also update documentation of `umfMemoryProviderAlloc`. 

We should either require `alignment` to always be a multiple of what `umfMemoryProviderGetMinPageSize()` returns (and return INVALID_ARGUMENT otherwise) or change the meaning of `alignment` param to `minAlignment` (and for example allow providers to return 8k aligned memory when 6k alignment is requested). @vinser52 any preferences?

Reply:
IMHO the first option is better:
"We should require `alignment` to always be a multiple of what `umfMemoryProviderGetMinPageSize()` returns (and return `INVALID_ARGUMENT` otherwise) "

Reply:
I prefer the first one since it is more clear. And it is consistent with mmap behavior.

Reply:
I think that the `alignment` should be a multiple of the minimum page size (`umfMemoryProviderGetMinPageSize()`) **or a divider of that**. If it is a divider, the address will be aligned to the minimum page size (for example: alignment 1024 is also a correct alignment, because an address aligned to 4096 is also aligned to 1024).

Reply:
`alignment` is verified in `os_alloc()` now, 
see: 
- https://github.com/oneapi-src/unified-memory-framework/pull/91
- https://github.com/oneapi-src/unified-memory-framework/pull/90

Done.

Reply:
> I think that the alignment should be a multiple of the minimum page size (umfMemoryProviderGetMinPageSize()) or a divider of that. If it is a divider, the address will be aligned to the minimum page size (for example: alignment 1024 is also a correct alignment, because an address aligned to 4096 is also aligned to 1024).

Ok, but that's a bit inconsistent IMO. Why is 1024 a correct alignment and 6k is not? I could argue that since an address aligned to 8k (a multiple of page size) is also aligned to 6k it should be OK as well. A generic code that wants to wrap our allocate would have to have a logic like this (or just always round up to page size): 

```
assert(is_pow2(alignment);
if (alignment < `umfMemoryProviderGetMinPageSize()`) return umfMemoryProviderAlloc(..., alignment);
else return umfMemoryProviderAlloc(..., round_up_to_page(alignment));
```

I would prefer either option no 1 (both 1k and 6k is disallowed) or option 2 (both are fine).

Reply:
- 6k is not a power of 2 and you have `assert(is_pow2(alignment);` here, so it would assert in case of 6k :-)
- all dividers of 4k (for example 1024) are powers of 2

Reply:
Right, the assert in that example would fail ;d

Right, being a power of 2 and 'being a divider or a multiple of the page size' is always equivalent so from that perspective I think the logic you proposed makes sense.